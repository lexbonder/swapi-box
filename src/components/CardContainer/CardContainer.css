.CardContainer h1 {
  text-align: center;
  color: #feda4a;
}

.CardContainer section {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}