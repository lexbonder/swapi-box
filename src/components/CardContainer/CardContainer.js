import React from 'react';
import Card from '../Card/Card'
// import People from '../People/People'; 

const CardContainer = ({chosenCategory, cleanedData}) => {
  const cardsToRender = cleanedData.map((datapoint, index) => 
    <Card key={index} cardInfo={datapoint} category={chosenCategory} /> )

  return(
    <div>
      <h1>{chosenCategory}</h1>
      {cardsToRender}
    </div>
  )
}

export default CardContainer;