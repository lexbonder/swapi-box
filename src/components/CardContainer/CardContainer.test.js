import React from 'react';
import {shallow} from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {

  it('should match the snapshot if it recieves no favorites', () => {
    const wrapper = shallow(<CardContainer
      chosenCategory={'favorites'}
      cleanedData={[]}
      toggleFavorite={()=>{}}
      currentFavorites={[]}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if it recieves favorites', () => {
    const wrapper = shallow(<CardContainer
      chosenCategory={'favorites'}
      cleanedData={[{}, {}, {}]}
      toggleFavorite={()=>{}}
      currentFavorites={[]}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if it recieves people', () => {
    const wrapper = shallow(<CardContainer
      chosenCategory={'people'}
      cleanedData={[{}, {}, {}]}
      toggleFavorite={()=>{}}
      currentFavorites={[]}
    />)

    expect(wrapper).toMatchSnapshot()
  })

})