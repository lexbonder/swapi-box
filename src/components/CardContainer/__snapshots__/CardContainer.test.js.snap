// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot if it recieves favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    chosenCategory="favorites"
    cleanedData={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          favorites
</h1>,
        <div />,
        Array [
          <Card
            cardInfo={Object {}}
            toggleFavorite={[Function]}
/>,
          <Card
            cardInfo={Object {}}
            toggleFavorite={[Function]}
/>,
          <Card
            cardInfo={Object {}}
            toggleFavorite={[Function]}
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorites",
        },
        "ref": null,
        "rendered": "favorites",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {},
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {},
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {},
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            favorites
</h1>,
          <div />,
          Array [
            <Card
              cardInfo={Object {}}
              toggleFavorite={[Function]}
/>,
            <Card
              cardInfo={Object {}}
              toggleFavorite={[Function]}
/>,
            <Card
              cardInfo={Object {}}
              toggleFavorite={[Function]}
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorites",
          },
          "ref": null,
          "rendered": "favorites",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {},
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {},
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {},
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot if it recieves no favorites 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    chosenCategory="favorites"
    cleanedData={Array []}
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          favorites
</h1>,
        <h1>
          No favorites found
</h1>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorites",
        },
        "ref": null,
        "rendered": "favorites",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No favorites found",
        },
        "ref": null,
        "rendered": "No favorites found",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            favorites
</h1>,
          <h1>
            No favorites found
</h1>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorites",
          },
          "ref": null,
          "rendered": "favorites",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No favorites found",
          },
          "ref": null,
          "rendered": "No favorites found",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CardContainer should match the snapshot if it recieves people 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    chosenCategory="people"
    cleanedData={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          people
</h1>,
        <div />,
        Array [
          <Card
            cardInfo={Object {}}
            toggleFavorite={[Function]}
/>,
          <Card
            cardInfo={Object {}}
            toggleFavorite={[Function]}
/>,
          <Card
            cardInfo={Object {}}
            toggleFavorite={[Function]}
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
        },
        "ref": null,
        "rendered": "people",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {},
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {},
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "cardInfo": Object {},
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            people
</h1>,
          <div />,
          Array [
            <Card
              cardInfo={Object {}}
              toggleFavorite={[Function]}
/>,
            <Card
              cardInfo={Object {}}
              toggleFavorite={[Function]}
/>,
            <Card
              cardInfo={Object {}}
              toggleFavorite={[Function]}
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
          },
          "ref": null,
          "rendered": "people",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {},
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {},
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "cardInfo": Object {},
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
